### PR TITLE
Add assumption about email titles

### LINKED ISSUE #1924: Can emails be "web pages (HTML)"?
We have a definition of [web pages (HTML)](https://act-rules.github.io/glossary/#web-page-html) stating:
> An HTML [web page](https://www.w3.org/TR/WCAG21/#dfn-web-page-s) is the set of all [fully active](https://html.spec.whatwg.org/#fully-active) [documents](https://dom.spec.whatwg.org/#concept-document) which share the same [top-level browsing context](https://html.spec.whatwg.org/#top-level-browsing-context).

I do feel this would include emails written as HTML, where the content is a document.

---
The problem? [the specs](https://html.spec.whatwg.org/multipage/semantics.html#the-head-element) have this wonderful note:
> The [title](https://html.spec.whatwg.org/multipage/semantics.html#the-title-element) element is a required child in most situations, but when a higher-level protocol provides [title](https://html.spec.whatwg.org/multipage/semantics.html#the-title-element) information, e.g., in the subject line of an email when HTML is used as an email authoring format, the title element can be omitted.

and I would agree that a document inside an email has its title provided by the message's subject and thus does not require a `title` element to satisfy 2.4.2… But the current rule would fail it…

---
Fortunately, that should be fairly easy to fix. The language rules have an assumption in the line of "`lang` is used, not a `content-language` HTTP header". We can similarly add an assumption to the title rules in the line of "`title` is used, not email subject".

### PR DESCRIPTION
**Pull request by Jym77**
Add an assumption about higher protocol providing title.
Vocabulary, and example, follow the note in [the specification of the `head` element](https://html.spec.whatwg.org/multipage/semantics.html#the-head-element):
> The [title](https://html.spec.whatwg.org/multipage/semantics.html#the-title-element) element is a required child in most situations, but when a higher-level protocol provides [title](https://html.spec.whatwg.org/multipage/semantics.html#the-title-element) information, e.g., in the subject line of an email when HTML is used as an email authoring format, the title element can be omitted

Closes issue(s):

- closes #1924

Need for Call for Review:
This will require a 1 week Call for Review (adding an assumption).

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/html-page-non-empty-title-2779a5.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ For each target element, the first [HTML][] `title` element that is a [descendan
 
 This rule assumes that [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG21/#page-titled) does not require that a document only has one `title` element, nor that it is a child of the `head` element of a document. While this is invalid in HTML, the HTML 5.2 specification describes what should happen in case of multiple titles, and titles outside the `head` element. Because of this, neither of these validation issues causes a conformance problem for WCAG. Regardless of whether this is required by 2.4.2 Page Titled, failing this rule means the success criterion is not satisfied.
 
+This rule assumes that the title of the page is not provided by a higher-level protocol. For example, the subject field of an email authored in HTML would provide a title without requiring a `title` element. In such a case, this rule will fail while [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG21/#page-titled) may still be satisfied.
```
Comment (carlosapaduarte) [inline _rules/html-page-non-empty-title-2779a5.md]:
```suggestion
This rule assumes that the title of the page is not provided by a higher-level protocol. For example, the subject field of an email authored in HTML can provide a title without requiring a `title` element. In such a case, this rule will fail while [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG21/#page-titled) may still be satisfied.
```

**Review by carlosapaduarte — APPROVED** [review]
Just an editorial suggestion, but approving already

**Review by giacomo-petri — APPROVED** [review]

**Comment by giacomo-petri** [comment]
@Jym77 ,

Considering the new "Secondary requirement" type, from

> Scenario 2: the rule's failed outcomes require further testing for the accessibility requirement

should we move the assumption to the "Background" section, adding "secondary: true" as parameter for this rule?

**Review by ajanec01 — APPROVED** [review]

**Comment by Jym77** [comment]
Call for review ends on January 16th.

**Review by WilcoFiers — APPROVED** [review]

**Comment by Jym77** [comment]
Call for Review has ended, merging.

**Pull request merged**